cpp/tests/table/experimental_row_operator_tests.cu

```diff
@@ -229,6 +260,148 @@ TYPED_TEST(TypedTableViewTest, TestLexicographicalComparatorSameTable)
+{
+  using data_col    = cudf::test::fixed_width_column_wrapper<TypeParam>;
+  using int32s_col  = cudf::test::fixed_width_column_wrapper<int32_t>;
+  using strings_col = cudf::test::strings_column_wrapper;
+  using structs_col = cudf::test::structs_column_wrapper;
+
+  auto const col1 = [] {
+    auto const get_structs = [] {
+      auto child0 = data_col{0, 3, 0, 2};
+      auto child1 = strings_col{"a", "c", "a", "b"};
+      return structs_col{{child0, child1}};
+    };
+    return cudf::make_lists_column(
+      2, int32s_col{0, 2, 4}.release(), get_structs().release(), 0, {});
+  }();
+  auto const col2 = [] {
+    auto const get_structs = [] {
+      auto child0 = data_col{};
+      auto child1 = strings_col{};
+      return structs_col{{child0, child1}};
+    };
+    return cudf::make_lists_column(0, int32s_col{}.release(), get_structs().release(), 0, {});
+  }();
+
+  auto const column_order = std::vector{cudf::order::ASCENDING};
+  auto const lhs          = cudf::table_view{{*col1}};
+  auto const rhs          = cudf::table_view{{*col2}};
+
+  auto const stream    = cudf::get_default_stream();
+  auto const test_sort = [stream](auto const& preprocessed,
+                                  auto const& input,
+                                  auto const& comparator,
+                                  auto const& expected) {
+    auto const order = sorted_order(
+      preprocessed, input.num_rows(), cudf::detail::has_nested_columns(input), comparator, stream);
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, order->view());
+  };
+
+  auto const test_sort_two_tables = [&](auto const& preprocessed_lhs,
+                                        auto const& preprocessed_rhs) {
+    auto const expected_lhs = int32s_col{1, 0};
+    test_sort(preprocessed_lhs,
+              lhs,
+              cudf::experimental::row::lexicographic::sorting_physical_element_comparator{},
+              expected_lhs);
+
+    auto const expected_rhs = int32s_col{};
+    test_sort(preprocessed_rhs,
+              rhs,
+              cudf::experimental::row::lexicographic::sorting_physical_element_comparator{},
+              expected_rhs);
+
+    EXPECT_THROW(test_sort(preprocessed_lhs,
+                           lhs,
+                           cudf::experimental::row::lexicographic::physical_element_comparator{},
+                           expected_lhs),
+                 cudf::logic_error);
+    EXPECT_THROW(test_sort(preprocessed_rhs,
+                           rhs,
+                           cudf::experimental::row::lexicographic::physical_element_comparator{},
+                           expected_rhs),
+                 cudf::logic_error);
+  };
+
+  // Generate preprocessed data for both lhs and lhs at the same time.
+  // Switching order of lhs and rhs tables then sorting them using their preprocessed data should
+  // produce exactly the same result.
+  {
+    auto const [preprocessed_lhs, preprocessed_rhs /*empty*/] =
```

Comment:
Just rename `[preprocess_]rhs` to `[preprocessed_]empty_rhs` rather than carrying around this comment everywhere.